lib/sycamore/sycamore/transforms/extract_graph.py

```diff
@@ -0,0 +1,159 @@
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Dict, Any
+from sycamore.plan_nodes import Node
+from sycamore.transforms.map import Map
+from sycamore.data import Document, MetadataDocument
+import uuid
+
+if TYPE_CHECKING:
+    from sycamore.docset import DocSet
+
+
+class GraphData(ABC):
+    def __init__(self):
+        pass
+
+
+class GraphMetadata(GraphData):
+    """
+    Object which handles what fields to extract metadata from and what fields to represent them as in neo4j
+
+    Args:
+        nodeKey: Key used to access document metadata in the document['properties] dictionary
+        nodeLabel: The label used in neo4j the node of a piece of metadata
+        relLabel: The label used in neo4j for the relationship between the document and a piece of metadata
```

Comment:
Is there a description of the type of entity to be extracted? Or are you relying only on the name (nodeLabel)?

Reply:
This is for the metadata extractor currently so it extracts by the key of the metadata provided in the initial jsonmanifest